src/transformers/trainer.py

```diff
@@ -1869,14 +1754,6 @@ def _inner_training_loop(
                     if args.max_grad_norm is not None and args.max_grad_norm > 0:
                         # deepspeed does its own clipping
 
-                        if self.do_grad_scaling:
-                            # Reduce gradients first for XLA
-                            if is_torch_tpu_available():
-                                gradients = xm._fetch_gradients(self.optimizer)
-                                xm.all_reduce("sum", gradients, scale=1.0 / xm.xrt_world_size())
-                            # AMP: gradients need unscaling
-                            self.scaler.unscale_(self.optimizer)
```

Comment:
cc @muellerzr as you have more idea about TPUs

Reply:
@muellerzr Could you please take a look at these modifications？Thanks :-)

Reply:
This XLA logic needs to be maintained in here, or have the equivalent done in Accelerate in terms of the `all_reduce` + `fetch_gradients`. If you'd like to open a PR for this that'd be phenomenal! (Sorry, had to go dig in accelerate for a bit to find the relevant areas)

Reply:
Once I figure out how it works, I would be happy to have the equivalent done in Accelerate, but it might take some time :-).

Reply:
remove XLA logic, since Accelerate will handle it automatically.

Reply:
This has been merged into main on Accelerate via https://github.com/huggingface/accelerate/pull/1926